query/src/sql/plan_parser.rs

```diff
@@ -133,13 +134,29 @@ impl PlanParser {
             DfStatement::ShowCreateTable(v) => self.sql_show_create_table_to_plan(v),
 
             // TODO: support like and other filters in show queries
-            DfStatement::ShowTables(_) => self.build_from_sql(
-                format!(
-                    "SELECT name FROM system.tables where database = '{}' ORDER BY database, name",
-                    self.ctx.get_current_database()
-                )
-                .as_str(),
-            ),
+            DfStatement::ShowTables(df) => {
+                let show_sql = match df {
+                    DfShowTables::All => {
+                        format!(
+                            "SELECT name FROM system.tables where database = '{}' ORDER BY database, name",
+                            self.ctx.get_current_database()
+                        )
+                    }
+                    DfShowTables::Like(i) => {
+                        format!(
+                            "SELECT name FROM system.tables where database = '{}' AND name LIKE {} ORDER BY database, name",
+                            self.ctx.get_current_database(), i,
+                        )
+                    }
+                    DfShowTables::Where(e) => {
```

Comment:
What happens to the following SQL?
```
SHOW TABLES WHERE 1 = 1  OR 1 = 1 
```

Reply:
> What happens to the following SQL?
> 
> ```
> SHOW TABLES WHERE 1 = 1  OR 1 = 1 
> ```

It's ok, we don't care about it, just let it be any complex expr. The optimize / select executor will handle that.

Reply:
According to MySQL standard,`parse_expr` can handle all of them: https://dev.mysql.com/doc/refman/5.7/en/expressions.html

Buf I didn't check the code in sqlparse-rs, so I cannot guarantee it works well.

Reply:
May this change the behavior of the query?

After format: 
`SELECT name FROM system.tables where database = '{}' AND 1 = 1 OR 1 = 1 ORDER BY database, name`

Reply:
> May this change the behavior of the query?
> 
> After format:
> `SELECT name FROM system.tables where database = '{}' AND 1 = 1 OR 1 = 1 ORDER BY database, name`

Got It. Sorry for misunderstanding this case! I will adding tests about it in stateless test.

Reply:
> evaluation proceeds left to right, with the exception that assignments evaluate right to left

According to https://dev.mysql.com/doc/refman/8.0/en/operator-precedence.html , it will not return all tables. But I think maybe it's proper to build sql like:
```
SELECT name FROM system.tables where database = '{}' AND (1 = 1 OR 1 = 1) ORDER BY database, name
```

@zhang2014 do you think it's ok?

Reply:
> @zhang2014 do you think it's ok?

It looks ok